src/components/BarSeries.tsx

```diff
@@ -37,22 +37,26 @@ export function BarSeries(props: BarSeriesProps) {
       hidden: !displayMarker,
     };
   }, [color, displayMarker, figure]);
-
+  const visibility = useMemo(() => {
+    const value = legendState.labels.find((label) => label.id === id);
+    return value ? value.visibility : true;
+  }, [id, legendState.labels]);
   useEffect(() => {
     legendDispatch({
       type: 'ADD_LEGEND_LABEL',
       payload: {
         id,
+        visibility,
```

Comment:
We do not need to pass `isVisible` here. The default value should be `true` when the label is added.

Reply:
this dispatch is used to update every moment so if i set it to true the TOGGLE_VISIBILITY won't work correctly

Reply:
Only TOGGLE_VISIBILITY should modify `isVisible` you can keep the current state value when `ADD_LEGEND_LABEL` is handled.

Reply:
same problem